tests/test_controller_ticker.cpp

```diff
@@ -51,18 +51,46 @@ BOOST_AUTO_TEST_CASE(RUN)
       initq.push_back(qi);
     }
   }
-  controller.setEncoderValues(initq);
-  controller.init(initq);
+
+  std::vector<double> qEnc(initq.size(), 0);
+  std::vector<double> alphaEnc(initq.size(), 0);
+  auto simulateSensors = [&, qEnc, alphaEnc]() mutable {
+    auto & robot = controller.robot();
+    for(unsigned i = 0; i < robot.refJointOrder().size(); i++)
+    {
+      auto jIdx = robot.jointIndexInMBC(i);
+      if(jIdx != -1)
+      {
+        auto jointIndex = static_cast<unsigned>(jIdx);
+        qEnc[i] = robot.mbc().q[jointIndex][0];
+        alphaEnc[i] = robot.mbc().alpha[jointIndex][0];
+      }
+    }
+    controller.setEncoderValues(qEnc);
+    controller.setEncoderVelocities(alphaEnc);
+    controller.setSensorPositions({{"FloatingBase", robot.posW().translation()}});
+    controller.setSensorOrientations({{"FloatingBase", Eigen::Quaterniond{robot.posW().rotation()}}});
+  };
+
+  // Initialize state of robot, but not the controller as the observer will need
+  // the simulated sensors
+  controller.init(initq, false);
```

Comment:
I don't think we should introduce this extra boolean parameter for the sake of tests. In that case:
- the initial encoders values are `initq`
- the initial encoder velocities are zero
- the initial base position and orientation can be obtained from the `controller.robot().module().default_attitude()`

Reply:
Yes agreed. Initially I thought that the intialization interface decoupled the loading of the robot state from the actual controller initialization, but I discovered this wasn't the case when writing the test, and tried to change it in a way that does not affect existing code.

But I agree that the added boolean can be a bit confusing, and may also be misused. I'll revert this and do it inside the test ticker directly instead.